diff --git a/Makefile.am b/Makefile.am
index aa92d2e30..769813cce 100644
--- a/Makefile.am
+++ b/Makefile.am
@@ -23,9 +23,6 @@ comma := ,
 # Helper function to strip src/ and test/hs/ from a list
 strip_hsroot = $(patsubst src/%,%,$(patsubst test/hs/%,%,$(1)))
 
-# Use bash in order to be able to use pipefail
-SHELL=/bin/bash
-
 EXTRA_DIST=
 
 # Enable colors in shelltest
@@ -2849,9 +2846,6 @@ dist-release: dist
 	done
 
 install-exec-local:
-	@mkdir_p@ "$(DESTDIR)${localstatedir}/lib/ganeti" \
-	  "$(DESTDIR)${localstatedir}/log/ganeti" \
-	  "$(DESTDIR)${localstatedir}/run/ganeti"
 	for dir in $(SYMLINK_TARGET_DIRS); do \
 	  @mkdir_p@  $(DESTDIR)$$dir; \
 	done
@@ -3056,7 +3050,7 @@ ganeti.cabal: cabal/ganeti.template.cabal Makefile cabal/cabal-from-modules.py $
 # Target that builds all binaries (including those that are not
 # rebuilt except when running the tests)
 .PHONY: really-all
-really-all: all $(check_SCRIPTS) $(haskell_tests) $(HS_ALL_PROGS)
+really-all: all $(HS_ALL_PROGS) $(BUILT_EXAMPLES)
 
 # we don't need the ancient implicit rules:
 %: %,v
