diff --git a/lib/hypervisor/hv_kvm/monitor.py b/lib/hypervisor/hv_kvm/monitor.py
index 5ceca3e69..aca2a60f7 100644
--- a/lib/hypervisor/hv_kvm/monitor.py
+++ b/lib/hypervisor/hv_kvm/monitor.py
@@ -345,7 +345,7 @@ class QmpConnection(QemuMonitorSocket):
   """Connection to the QEMU Monitor using the QEMU Monitor Protocol (QMP).
 
   """
-  _QMP_TIMEOUT = 5
+  _QMP_TIMEOUT = 15
   _FIRST_MESSAGE_KEY = "QMP"
   _RETURN_KEY = "return"
   _ACTUAL_KEY = ACTUAL_KEY = "actual"
