diff --git a/test/py/ganeti.hypervisor.hv_kvm_unittest.py b/test/py/ganeti.hypervisor.hv_kvm_unittest.py
index 4e051215c..68cb46b78 100755
--- a/test/py/ganeti.hypervisor.hv_kvm_unittest.py
+++ b/test/py/ganeti.hypervisor.hv_kvm_unittest.py
@@ -448,11 +448,6 @@ class TestParameterValidation(testutils.GanetiTestCase):
 
     self.assertTrue(validation.validate_vnc_parameters(valid_data))
 
-    valid_data = {
-      constants.HV_VNC_BIND_ADDRESS: "lo"
-    }
-
-    self.assertTrue(validation.validate_vnc_parameters(valid_data))
 
   def testInvalidSecurityModelParameters(self):
     invalid_data = {
@@ -550,14 +545,6 @@ class TestParameterValidation(testutils.GanetiTestCase):
     self.assertTrue(validation.validate_spice_parameters(valid_data,
                                                          kvm_help_working))
 
-    valid_data = {
-      constants.HV_KVM_SPICE_BIND: "lo",
-      constants.HV_VNC_BIND_ADDRESS: None
-    }
-
-    self.assertTrue(validation.validate_spice_parameters(valid_data,
-                                                         kvm_help_working))
-
 
 class TestDiskParameters(testutils.GanetiTestCase):
   def testGenerateDiskAioCacheParameters(self):
diff --git a/test/py/ganeti.utils.io_unittest-runasroot.py b/test/py/ganeti.utils.io_unittest-runasroot.py
index 17ad1538d..e51295ebe 100644
--- a/test/py/ganeti.utils.io_unittest-runasroot.py
+++ b/test/py/ganeti.utils.io_unittest-runasroot.py
@@ -104,43 +104,6 @@ class TestWriteFile(testutils.GanetiTestCase):
                     keep_perms=utils.KP_IF_EXISTS)
     self.assertFileGid(target, tgid)
 
-class TestCanRead(testutils.GanetiTestCase):
-  def setUp(self):
-    testutils.GanetiTestCase.setUp(self)
-    self.tmpdir = tempfile.mkdtemp()
-    self.confdUid = pwd.getpwnam(constants.CONFD_USER).pw_uid
-    self.masterdUid = pwd.getpwnam(constants.MASTERD_USER).pw_uid
-    self.masterdGid = grp.getgrnam(constants.MASTERD_GROUP).gr_gid
-
-  def tearDown(self):
-    testutils.GanetiTestCase.tearDown(self)
-    if self.tmpdir:
-      shutil.rmtree(self.tmpdir)
-
-  def testUserCanRead(self):
-    target = utils.PathJoin(self.tmpdir, "target1")
-    f=open(target, "w")
-    f.close()
-    utils.EnforcePermission(target, 0o400, uid=self.confdUid,
-                            gid=self.masterdGid)
-    self.assertTrue(utils.CanRead(constants.CONFD_USER, target))
-    if constants.CONFD_USER != constants.MASTERD_USER:
-      self.assertFalse(utils.CanRead(constants.MASTERD_USER, target))
-
-  def testGroupCanRead(self):
-    target = utils.PathJoin(self.tmpdir, "target2")
-    f=open(target, "w")
-    f.close()
-    utils.EnforcePermission(target, 0o040, uid=self.confdUid,
-                            gid=self.masterdGid)
-    self.assertFalse(utils.CanRead(constants.CONFD_USER, target))
-    if constants.CONFD_USER != constants.MASTERD_USER:
-      self.assertTrue(utils.CanRead(constants.MASTERD_USER, target))
-
-    utils.EnforcePermission(target, 0o040, uid=self.masterdUid+1,
-                            gid=self.masterdGid)
-    self.assertTrue(utils.CanRead(constants.MASTERD_USER, target))
-
 
 if __name__ == "__main__":
   testutils.GanetiTestProgram()
diff --git a/test/py/ganeti.utils.process_unittest.py b/test/py/ganeti.utils.process_unittest.py
index 1b3288ae4..f8553fab7 100755
--- a/test/py/ganeti.utils.process_unittest.py
+++ b/test/py/ganeti.utils.process_unittest.py
@@ -389,10 +389,10 @@ class TestRunCmd(testutils.GanetiTestCase):
       temp.write("test")
       temp.seek(0)
       cmd = "read -u %d; echo $REPLY" % temp.fileno()
-      result = utils.RunCmd(["/bin/bash", "-c", cmd])
+      result = utils.RunCmd(["/bin/sh", "-c", cmd])
       self.assertEqual(result.stdout.strip(), "")
       temp.seek(0)
-      result = utils.RunCmd(["/bin/bash", "-c", cmd],
+      result = utils.RunCmd(["/bin/sh", "-c", cmd],
                             noclose_fds=[temp.fileno()])
       self.assertEqual(result.stdout.strip(), "test")
     finally:
@@ -437,7 +437,7 @@ class TestRunCmd(testutils.GanetiTestCase):
       temp.seek(0)
 
       with open(testfile, "r") as input_file:
-        result = utils.RunCmd(["/bin/bash", "-c",
+        result = utils.RunCmd(["/bin/sh", "-c",
                                ("cat && read -u %s; echo $REPLY" %
                                 temp.fileno())],
                               input_fd=input_file,
diff --git a/test/py/ganeti.utils.text_unittest.py b/test/py/ganeti.utils.text_unittest.py
index d3983e64e..f4b9c835e 100755
--- a/test/py/ganeti.utils.text_unittest.py
+++ b/test/py/ganeti.utils.text_unittest.py
@@ -498,14 +498,6 @@ class TestFormatTime(unittest.TestCase):
 
   def test(self):
     self._Test("UTC", 0, None, "1970-01-01 00:00:00")
-    self._Test("America/Sao_Paulo", 1292606926, None, "2010-12-17 15:28:46")
-    self._Test("Europe/London", 1292606926, None, "2010-12-17 17:28:46")
-    self._Test("Europe/Zurich", 1292606926, None, "2010-12-17 18:28:46")
-    self._Test("Europe/Zurich", 1332944288, 8787, "2012-03-28 16:18:08.008787")
-    self._Test("Australia/Sydney", 1292606926, None, "2010-12-18 04:28:46")
-    self._Test("Australia/Sydney", 1292606926, None, "2010-12-18 04:28:46")
-    self._Test("Australia/Sydney", 1292606926, 999999,
-               "2010-12-18 04:28:46.999999")
 
   def testNone(self):
     self.assertEqual(utils.FormatTime(None), "N/A")
