diff --git a/lib/ssh.py b/lib/ssh.py
index 994a0a309..6bad0da55 100644
--- a/lib/ssh.py
+++ b/lib/ssh.py
@@ -986,24 +986,6 @@ def _EnsureCorrectGanetiVersion(cmd):
 
   version = constants.DIR_VERSION
   all_cmds = [["test", "-d", os.path.join(pathutils.PKGLIBDIR, version)]]
-  if constants.HAS_GNU_LN:
-    all_cmds.extend([["ln", "-s", "-f", "-T",
-                      os.path.join(pathutils.PKGLIBDIR, version),
-                      os.path.join(pathutils.SYSCONFDIR, "ganeti/lib")],
-                     ["ln", "-s", "-f", "-T",
-                      os.path.join(pathutils.SHAREDIR, version),
-                      os.path.join(pathutils.SYSCONFDIR, "ganeti/share")]])
-  else:
-    all_cmds.extend([["rm", "-f",
-                      os.path.join(pathutils.SYSCONFDIR, "ganeti/lib")],
-                     ["ln", "-s", "-f",
-                      os.path.join(pathutils.PKGLIBDIR, version),
-                      os.path.join(pathutils.SYSCONFDIR, "ganeti/lib")],
-                     ["rm", "-f",
-                      os.path.join(pathutils.SYSCONFDIR, "ganeti/share")],
-                     ["ln", "-s", "-f",
-                      os.path.join(pathutils.SHAREDIR, version),
-                      os.path.join(pathutils.SYSCONFDIR, "ganeti/share")]])
   all_cmds.append(cmd)
   return all_cmds
 
