diff --git a/lib/hypervisor/hv_kvm/monitor.py b/lib/hypervisor/hv_kvm/monitor.py
index 5ceca3e69..9984775b4 100644
--- a/lib/hypervisor/hv_kvm/monitor.py
+++ b/lib/hypervisor/hv_kvm/monitor.py
@@ -725,7 +725,8 @@ class QmpConnection(QemuMonitorSocket):
     if self.version >= (3, 0, 0):
       arguments["max-postcopy-bandwidth"] = max_bandwidth
 
-    self.execute_qmp("migrate-set-parameters", arguments)
+    if self.version >= (2, 4, 0):
+      self.execute_qmp("migrate-set-parameters", arguments)
 
   @_ensure_connection
   def SetMigrationCapabilities(self, capabilities, state):
