diff --git a/Makefile.am b/Makefile.am
index e76c5e9be..ee54f70fc 100644
--- a/Makefile.am
+++ b/Makefile.am
@@ -113,15 +113,15 @@ iallocatorsdir = $(pkglibdir)/iallocators
 pytoolsdir = $(pkgpythondir)/tools
 docdir = $(versiondir)$(datadir)/doc/$(PACKAGE)
 ifupdir = $(sysconfdir)/ganeti
-NET_MODES = bridged ovs ovn routed
+NET_MODES = bridged openvswitch ovn routed
 
 NET_SCRIPT_LIST = \
 	tools/net-bridged-down \
 	tools/net-bridged-up \
 	tools/net-ovn-down \
 	tools/net-ovn-up \
-	tools/net-ovs-down \
-	tools/net-ovs-up \
+	tools/net-openvswitch-down \
+	tools/net-openvswitch-up \
 	tools/net-routed-down \
 	tools/net-routed-up
 
@@ -136,7 +136,7 @@ SYMLINK_TARGET_DIRS = \
 	$(libdir)/ganeti/iallocators \
 	$(libdir)/ganeti/network/bridged \
 	$(libdir)/ganeti/network/ovn \
-	$(libdir)/ganeti/network/ovs \
+	$(libdir)/ganeti/network/openvswitch \
 	$(libdir)/ganeti/network/routed \
 	$(libdir)/ganeti/tools \
 	$(prefix)/share/ganeti \
diff --git a/tools/net-openvswitch-down.in b/tools/net-openvswitch-down.in
new file mode 100644
index 000000000..8fb5f7c81
--- /dev/null
+++ b/tools/net-openvswitch-down.in
@@ -0,0 +1,33 @@
+#!/bin/bash
+#
+
+# Copyright (C) 2026 the Ganeti project
+# All rights reserved.
+#
+# Redistribution and use in source and binary forms, with or without
+# modification, are permitted provided that the following conditions are
+# met:
+#
+# 1. Redistributions of source code must retain the above copyright notice,
+# this list of conditions and the following disclaimer.
+#
+# 2. Redistributions in binary form must reproduce the above copyright
+# notice, this list of conditions and the following disclaimer in the
+# documentation and/or other materials provided with the distribution.
+#
+# THIS SOFTWARE IS PROVIDED BY THE COPYRIGHT HOLDERS AND CONTRIBUTORS "AS
+# IS" AND ANY EXPRESS OR IMPLIED WARRANTIES, INCLUDING, BUT NOT LIMITED
+# TO, THE IMPLIED WARRANTIES OF MERCHANTABILITY AND FITNESS FOR A PARTICULAR
+# PURPOSE ARE DISCLAIMED. IN NO EVENT SHALL THE COPYRIGHT HOLDER OR
+# CONTRIBUTORS BE LIABLE FOR ANY DIRECT, INDIRECT, INCIDENTAL, SPECIAL,
+# EXEMPLARY, OR CONSEQUENTIAL DAMAGES (INCLUDING, BUT NOT LIMITED TO,
+# PROCUREMENT OF SUBSTITUTE GOODS OR SERVICES; LOSS OF USE, DATA, OR
+# PROFITS; OR BUSINESS INTERRUPTION) HOWEVER CAUSED AND ON ANY THEORY OF
+# LIABILITY, WHETHER IN CONTRACT, STRICT LIABILITY, OR TORT (INCLUDING
+# NEGLIGENCE OR OTHERWISE) ARISING IN ANY WAY OUT OF THE USE OF THIS
+# SOFTWARE, EVEN IF ADVISED OF THE POSSIBILITY OF SUCH DAMAGE.
+
+source @PKGLIBDIR@/net-common
+
+# add your own network logic here
+
diff --git a/tools/net-openvswitch-up.in b/tools/net-openvswitch-up.in
new file mode 100644
index 000000000..2ac02f820
--- /dev/null
+++ b/tools/net-openvswitch-up.in
@@ -0,0 +1,47 @@
+#!/bin/bash
+#
+
+# Copyright (C) 2026 the Ganeti project
+# All rights reserved.
+#
+# Redistribution and use in source and binary forms, with or without
+# modification, are permitted provided that the following conditions are
+# met:
+#
+# 1. Redistributions of source code must retain the above copyright notice,
+# this list of conditions and the following disclaimer.
+#
+# 2. Redistributions in binary form must reproduce the above copyright
+# notice, this list of conditions and the following disclaimer in the
+# documentation and/or other materials provided with the distribution.
+#
+# THIS SOFTWARE IS PROVIDED BY THE COPYRIGHT HOLDERS AND CONTRIBUTORS "AS
+# IS" AND ANY EXPRESS OR IMPLIED WARRANTIES, INCLUDING, BUT NOT LIMITED
+# TO, THE IMPLIED WARRANTIES OF MERCHANTABILITY AND FITNESS FOR A PARTICULAR
+# PURPOSE ARE DISCLAIMED. IN NO EVENT SHALL THE COPYRIGHT HOLDER OR
+# CONTRIBUTORS BE LIABLE FOR ANY DIRECT, INDIRECT, INCIDENTAL, SPECIAL,
+# EXEMPLARY, OR CONSEQUENTIAL DAMAGES (INCLUDING, BUT NOT LIMITED TO,
+# PROCUREMENT OF SUBSTITUTE GOODS OR SERVICES; LOSS OF USE, DATA, OR
+# PROFITS; OR BUSINESS INTERRUPTION) HOWEVER CAUSED AND ON ANY THEORY OF
+# LIABILITY, WHETHER IN CONTRACT, STRICT LIABILITY, OR TORT (INCLUDING
+# NEGLIGENCE OR OTHERWISE) ARISING IN ANY WAY OUT OF THE USE OF THIS
+# SOFTWARE, EVEN IF ADVISED OF THE POSSIBILITY OF SUCH DAMAGE.
+
+source @PKGLIBDIR@/net-common
+
+# Remove stale port
+ovs-vsctl del-port $INTERFACE || true
+# Bring interface up
+ip link set $INTERFACE up
+# Add port
+ovs-vsctl add-port ${LINK} $INTERFACE
+# Set up access port
+# From gnt-instance man page vlan should be either .VLAN_ID or VLAN_ID
+ACPORT=${VLAN%%:*}  # remove any trunk info
+[ -n "$ACPORT" ] && ovs-vsctl set port $INTERFACE tag=${ACPORT#.}
+# Set up trunk port
+# From gnt-instance man page vlan should be :VLAN_ID[:VLAN_ID2..]
+TRUNKS=${VLAN#.*:}  # remove any access info
+TRUNKS=${TRUNKS#:}  # remove leading ':', if still present
+[ -n "$TRUNKS" ] && ovs-vsctl set port $INTERFACE trunks=${TRUNKS//:/,}
+
diff --git a/tools/net-ovs-down.in b/tools/net-ovs-down.in
deleted file mode 100644
index 8fb5f7c81..000000000
--- a/tools/net-ovs-down.in
+++ /dev/null
@@ -1,33 +0,0 @@
-#!/bin/bash
-#
-
-# Copyright (C) 2026 the Ganeti project
-# All rights reserved.
-#
-# Redistribution and use in source and binary forms, with or without
-# modification, are permitted provided that the following conditions are
-# met:
-#
-# 1. Redistributions of source code must retain the above copyright notice,
-# this list of conditions and the following disclaimer.
-#
-# 2. Redistributions in binary form must reproduce the above copyright
-# notice, this list of conditions and the following disclaimer in the
-# documentation and/or other materials provided with the distribution.
-#
-# THIS SOFTWARE IS PROVIDED BY THE COPYRIGHT HOLDERS AND CONTRIBUTORS "AS
-# IS" AND ANY EXPRESS OR IMPLIED WARRANTIES, INCLUDING, BUT NOT LIMITED
-# TO, THE IMPLIED WARRANTIES OF MERCHANTABILITY AND FITNESS FOR A PARTICULAR
-# PURPOSE ARE DISCLAIMED. IN NO EVENT SHALL THE COPYRIGHT HOLDER OR
-# CONTRIBUTORS BE LIABLE FOR ANY DIRECT, INDIRECT, INCIDENTAL, SPECIAL,
-# EXEMPLARY, OR CONSEQUENTIAL DAMAGES (INCLUDING, BUT NOT LIMITED TO,
-# PROCUREMENT OF SUBSTITUTE GOODS OR SERVICES; LOSS OF USE, DATA, OR
-# PROFITS; OR BUSINESS INTERRUPTION) HOWEVER CAUSED AND ON ANY THEORY OF
-# LIABILITY, WHETHER IN CONTRACT, STRICT LIABILITY, OR TORT (INCLUDING
-# NEGLIGENCE OR OTHERWISE) ARISING IN ANY WAY OUT OF THE USE OF THIS
-# SOFTWARE, EVEN IF ADVISED OF THE POSSIBILITY OF SUCH DAMAGE.
-
-source @PKGLIBDIR@/net-common
-
-# add your own network logic here
-
diff --git a/tools/net-ovs-up.in b/tools/net-ovs-up.in
deleted file mode 100644
index 2ac02f820..000000000
--- a/tools/net-ovs-up.in
+++ /dev/null
@@ -1,47 +0,0 @@
-#!/bin/bash
-#
-
-# Copyright (C) 2026 the Ganeti project
-# All rights reserved.
-#
-# Redistribution and use in source and binary forms, with or without
-# modification, are permitted provided that the following conditions are
-# met:
-#
-# 1. Redistributions of source code must retain the above copyright notice,
-# this list of conditions and the following disclaimer.
-#
-# 2. Redistributions in binary form must reproduce the above copyright
-# notice, this list of conditions and the following disclaimer in the
-# documentation and/or other materials provided with the distribution.
-#
-# THIS SOFTWARE IS PROVIDED BY THE COPYRIGHT HOLDERS AND CONTRIBUTORS "AS
-# IS" AND ANY EXPRESS OR IMPLIED WARRANTIES, INCLUDING, BUT NOT LIMITED
-# TO, THE IMPLIED WARRANTIES OF MERCHANTABILITY AND FITNESS FOR A PARTICULAR
-# PURPOSE ARE DISCLAIMED. IN NO EVENT SHALL THE COPYRIGHT HOLDER OR
-# CONTRIBUTORS BE LIABLE FOR ANY DIRECT, INDIRECT, INCIDENTAL, SPECIAL,
-# EXEMPLARY, OR CONSEQUENTIAL DAMAGES (INCLUDING, BUT NOT LIMITED TO,
-# PROCUREMENT OF SUBSTITUTE GOODS OR SERVICES; LOSS OF USE, DATA, OR
-# PROFITS; OR BUSINESS INTERRUPTION) HOWEVER CAUSED AND ON ANY THEORY OF
-# LIABILITY, WHETHER IN CONTRACT, STRICT LIABILITY, OR TORT (INCLUDING
-# NEGLIGENCE OR OTHERWISE) ARISING IN ANY WAY OUT OF THE USE OF THIS
-# SOFTWARE, EVEN IF ADVISED OF THE POSSIBILITY OF SUCH DAMAGE.
-
-source @PKGLIBDIR@/net-common
-
-# Remove stale port
-ovs-vsctl del-port $INTERFACE || true
-# Bring interface up
-ip link set $INTERFACE up
-# Add port
-ovs-vsctl add-port ${LINK} $INTERFACE
-# Set up access port
-# From gnt-instance man page vlan should be either .VLAN_ID or VLAN_ID
-ACPORT=${VLAN%%:*}  # remove any trunk info
-[ -n "$ACPORT" ] && ovs-vsctl set port $INTERFACE tag=${ACPORT#.}
-# Set up trunk port
-# From gnt-instance man page vlan should be :VLAN_ID[:VLAN_ID2..]
-TRUNKS=${VLAN#.*:}  # remove any access info
-TRUNKS=${TRUNKS#:}  # remove leading ':', if still present
-[ -n "$TRUNKS" ] && ovs-vsctl set port $INTERFACE trunks=${TRUNKS//:/,}
-
