diff --git a/lib/client/gnt_cluster.py b/lib/client/gnt_cluster.py
index c2e4e7f56..696eb02ee 100644
--- a/lib/client/gnt_cluster.py
+++ b/lib/client/gnt_cluster.py
@@ -1169,12 +1169,6 @@ def _RenewCrypto(new_cluster_cert, new_rapi_cert, # pylint: disable=R0911
       ctx.ssh.CopyFileToNode(node_name, port,
                              ssconf_master_candidate_certs_filename)
 
-    # Write the boostrap entry to the config using wconfd.
-    config_live_lock = utils.livelock.LiveLock("renew_crypto")
-    cfg = config.GetConfig(None, config_live_lock)
-    cfg.AddNodeToCandidateCerts(constants.CRYPTO_BOOTSTRAP, master_digest)
-    cfg.Update(cfg.GetClusterInfo(), ctx.feedback_fn)
-
   def _RenewServerAndClientCerts(ctx):
     ctx.feedback_fn("Updating the cluster SSL certificate.")
 
