diff --git a/Makefile.am b/Makefile.am
index 02fd0a478..9589a568b 100644
--- a/Makefile.am
+++ b/Makefile.am
@@ -23,9 +23,6 @@ comma := ,
 # Helper function to strip src/ and test/hs/ from a list
 strip_hsroot = $(patsubst src/%,%,$(patsubst test/hs/%,%,$(1)))
 
-# Use bash in order to be able to use pipefail
-SHELL=/bin/bash
-
 EXTRA_DIST=
 
 # Enable colors in shelltest
@@ -1935,7 +1932,6 @@ python_tests_legacy = \
 	test/py/legacy/cmdlib/test_unittest.py \
 	test/py/legacy/cfgupgrade_unittest.py \
 	test/py/legacy/docs_unittest.py \
-	test/py/legacy/ganeti.asyncnotifier_unittest.py \
 	test/py/legacy/ganeti.backend_unittest-runasroot.py \
 	test/py/legacy/ganeti.backend_unittest.py \
 	test/py/legacy/ganeti.bootstrap_unittest.py \
@@ -1948,7 +1944,6 @@ python_tests_legacy = \
 	test/py/legacy/ganeti.confd.client_unittest.py \
 	test/py/legacy/ganeti.config_unittest.py \
 	test/py/legacy/ganeti.constants_unittest.py \
-	test/py/legacy/ganeti.daemon_unittest.py \
 	test/py/legacy/ganeti.errors_unittest.py \
 	test/py/legacy/ganeti.hooks_unittest.py \
 	test/py/legacy/ganeti.ht_unittest.py \
@@ -1992,7 +1987,6 @@ python_tests_legacy = \
 	test/py/legacy/ganeti.storage.filestorage_unittest.py \
 	test/py/legacy/ganeti.storage.gluster_unittest.py \
 	test/py/legacy/ganeti.tools.burnin_unittest.py \
-	test/py/legacy/ganeti.tools.ensure_dirs_unittest.py \
 	test/py/legacy/ganeti.tools.node_daemon_setup_unittest.py \
 	test/py/legacy/ganeti.tools.prepare_node_join_unittest.py \
 	test/py/legacy/ganeti.uidpool_unittest.py \
@@ -2874,9 +2868,6 @@ dist-release: dist
 	done
 
 install-exec-local:
-	@mkdir_p@ "$(DESTDIR)${localstatedir}/lib/ganeti" \
-	  "$(DESTDIR)${localstatedir}/log/ganeti" \
-	  "$(DESTDIR)${localstatedir}/run/ganeti"
 	for dir in $(SYMLINK_TARGET_DIRS); do \
 	  @mkdir_p@  $(DESTDIR)$$dir; \
 	done
@@ -3042,7 +3033,7 @@ dist/setup-config: ganeti.cabal $(HS_BUILT_SRCS)
 # Target that builds all binaries (including those that are not
 # rebuilt except when running the tests)
 .PHONY: really-all
-really-all: all $(check_SCRIPTS) $(haskell_tests) $(HS_ALL_PROGS)
+really-all: all $(check_SCRIPTS) $(haskell_tests) $(HS_ALL_PROGS) $(BUILT_EXAMPLES)
 
 # we don't need the ancient implicit rules:
 %: %,v
